src/store.tsx

```diff
@@ -0,0 +1,117 @@
+import { useReducer, useEffect, createContext, useContext, useCallback, useMemo, useState } from "react";
+
+export interface Movie {
+  id: string;
+  name: string;
+  genres: string[];
+  year: number;
+  poster: string;
+}
+
+function useMoviesSource(): {
+  movies: Movie[];
+  filterBy: string;
+  setFilterBy: (filterBy: string) => void;
+  sortBy: string;
+  setSortBy: (sortBy: string) => void;
+  selectedMovie: Movie | undefined;
+  setSelectedMovieId: (movieId: string) => void;
+} {
+  type MoviesState = {
+    movies: Movie[];
+    filterBy: string;
+    sortBy: string;
+  };
+
+  type MoviesAction =
+    | { type: "setMovies"; payload: Movie[] }
+    | { type: "setFilterBy"; payload: string }
+    | { type: "setSortBy"; payload: string };
```

Comment:
Please, do not define types within a hook. Better move it outside, and better move it to a separate `interfaces.ts` file.
Also, it is convenient to have a folder like `src/store/`